packages/mui-material-next/src/Button/Button.js

```diff
@@ -0,0 +1,654 @@
+      padding: '7px 21px',
+      fontSize: theme.typography.pxToRem(15),
+    }),
+  ...(ownerState.size === 'small' &&
+    ownerState.variant === 'contained' && {
+      padding: '4px 10px',
+      fontSize: theme.typography.pxToRem(13),
+    }),
+  ...(ownerState.size === 'large' &&
+    ownerState.variant === 'contained' && {
+      padding: '8px 22px',
+      fontSize: theme.typography.pxToRem(15),
+    }),
+  ...(ownerState.fullWidth && {
+    width: '100%',
+  }),
+  ...(ownerState.disableElevation && {
+    boxShadow: 'none',
+    '&:hover': {
+      boxShadow: 'none',
+    },
+    [`&.${buttonClasses.focusVisible}`]: {
+      boxShadow: 'none',
+    },
+    '&:active': {
+      boxShadow: 'none',
+    },
+    [`&.${buttonClasses.disabled}`]: {
+      boxShadow: 'none',
+    },
+  }),
+}));
+
+const ButtonStartIcon = styled('span', {
+  name: 'MuiButton',
+  slot: 'StartIcon',
+  overridesResolver: (props, styles) => {
+    const { ownerState } = props;
+
+    return [styles.startIcon, styles[`iconSize${capitalize(ownerState.size)}`]];
+  },
+})(({ ownerState }) => ({
+  display: 'inherit',
+  marginRight: 8,
+  marginLeft: -4,
+  ...(ownerState.size === 'small' && {
+    marginLeft: -2,
+  }),
+  ...commonIconStyles(ownerState),
+}));
+
+const ButtonEndIcon = styled('span', {
+  name: 'MuiButton',
+  slot: 'EndIcon',
+  overridesResolver: (props, styles) => {
+    const { ownerState } = props;
+
+    return [styles.endIcon, styles[`iconSize${capitalize(ownerState.size)}`]];
+  },
+})(({ ownerState }) => ({
+  display: 'inherit',
+  marginRight: -4,
+  marginLeft: 8,
+  ...(ownerState.size === 'small' && {
+    marginRight: -2,
+  }),
+  ...commonIconStyles(ownerState),
+}));
+
+const Button = React.forwardRef(function Button(inProps, ref) {
+  const props = useThemeProps({ props: inProps, name: 'MuiButton' });
+  const {
+    action,
+    centerRipple = false,
+    children,
+    className,
+    color = 'primary',
+    component = 'button',
+    disabled = false,
+    disableElevation = false,
+    disableFocusRipple = false,
+    disableRipple = false,
+    disableTouchRipple = false,
+    endIcon: endIconProp,
+    focusVisibleClassName,
+    fullWidth = false,
+    LinkComponent = 'a',
+    onBlur,
+    onClick,
+    onContextMenu,
+    onDragLeave,
+    onFocus,
+    onFocusVisible,
+    onKeyDown,
+    onKeyUp,
+    onMouseDown,
+    onMouseLeave,
+    onMouseUp,
+    onTouchEnd,
+    onTouchMove,
+    onTouchStart,
+    size = 'medium',
+    startIcon: startIconProp,
+    tabIndex = 0,
+    TouchRippleProps,
+    type,
+    variant = 'text',
+    ...other
+  } = props;
+
+  const buttonRef = React.useRef(null);
+  const handleRef = useForkRef(buttonRef, ref);
+
+  const rippleRef = React.useRef(null);
+
+  let ComponentProp = component;
+
+  if (ComponentProp === 'button' && (other.href || other.to)) {
+    ComponentProp = LinkComponent;
+  }
+
+  const { focusVisible, setFocusVisible, getRootProps } = useButton({
+    ...props,
```

Comment:
Maybe we should use the `ownerState` here? Otherwise we would loose all defaults we are setting when destructuring

Reply:
Well, we can't, as `ownerState` has properties that `useButton` returns :)